File: app/src/main/java/nich/work/aequorea/Aequorea.java
package nich.work.aequorea;

import android.app.Application;
import android.util.Log;

import com.flurry.android.FlurryAgent;
import com.flurry.android.FlurryPerformance;
import com.zzhoujay.richtext.RichText;

import java.io.File;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import nich.work.aequorea.common.Constants;
import nich.work.aequorea.common.cache.ArticleCache;
import nich.work.aequorea.common.utils.ThemeHelper;

public class Aequorea extends Application {
    private static Aequorea mApp;
    private static String mCurrentTheme;
    private static ExecutorService mExecutor;
    
    @Override
    public void onCreate() {
        super.onCreate();
        
        mApp = this;
        mCurrentTheme = ThemeHelper.getTheme();
        mExecutor = Executors.newCachedThreadPool();
    
        initCache();
        initFlurry();
    }

    private void initFlurry() {
        new FlurryAgent.Builder()
                .withDataSaleOptOut(false) //CCPA - the default value is false
                .withCaptureUncaughtExceptions(true)
                .withIncludeBackgroundSessionsInMetrics(true)
                .withLogLevel(Log.VERBOSE)
                .withPerformanceMetrics(FlurryPerformance.ALL)
                .build(this, "CXH6FJ9VS4P8K3K8BFN9");
    }
    
    private void initCache() {
        File cacheDir = new File(getCacheDir().getAbsolutePath() + File.separator + Constants.ARTICLE_PIC_CACHE);
        RichText.initCacheDir(cacheDir);
        ArticleCache.getCache();
    }
    
    public static Application getApp() {
        return mApp;
    }
    
    public static String getCurrentTheme() {
        return mCurrentTheme;
    }
    
    public static boolean isLightTheme() {
        return mCurrentTheme.equals(Constants.THEME_LIGHT);
    }
    
    public static void setCurrentTheme(String theme) {
        mCurrentTheme = theme;
    }
    
    public static ExecutorService getExecutor(){
        return mExecutor;
    }
}

File: app/src/main/java/nich/work/aequorea/Aequorea.kt
package nich.work.aequorea

import android.app.Activity
import android.app.Application
import com.flurry.android.FlurryAgent
import com.zzhoujay.richtext.RichText
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import nich.work.aequorea.common.Constants
import nich.work.aequorea.common.utils.ThemeHelper
import nich.work.aequorea.common.utils.releaseOnly
import nich.work.aequorea.di.AppInjector
import java.io.File
import javax.inject.Inject

class Aequorea : Application(), HasActivityInjector {

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>

    companion object {
        lateinit var app: Aequorea
        lateinit var currentTheme: String

        fun isLightTheme(): Boolean {
            return currentTheme == Constants.THEME_LIGHT
        }
    }

    override fun onCreate() {
        super.onCreate()

        app = this
        currentTheme = ThemeHelper.getTheme()

        releaseOnly { FlurryAgent.Builder().build(this, BuildConfig.FLURRY_ID) }

        AppInjector.init(this)
        DebuggableProvider.init(this)

        initCache()
    }

    private fun initCache() {
        val cacheDir = File(cacheDir?.absolutePath + File.separator + Constants.ARTICLE_CACHE)
        RichText.initCacheDir(cacheDir)
    }

    override fun activityInjector(): AndroidInjector<Activity> {
        return dispatchingAndroidInjector
    }

}